pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test: TestRepeatableReadLogic_select_for_share failed

pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test.TestRepeatableReadLogic_select_for_share [failed](https://mesolite.cluster.engflow.com/invocations/default/28c0fc1e-f827-47e9-aabd-223e579ea779?testReportRun=10&testReportShard=10&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbC1yZXBlYXRhYmxlLXJlYWQ6bG9jYWwtcmVwZWF0YWJsZS1yZWFkX3Rlc3Q=) on master @ [0cb591927ad6b9b2d547d71552716c5515618cec](https://github.com/cockroachdb/cockroach/commits/0cb591927ad6b9b2d547d71552716c5515618cec):


```
=== RUN   TestRepeatableReadLogic_select_for_share
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestRepeatableReadLogic_select_for_share278845014
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:150: cluster virtualization disabled due to issue: #156124 (expected label: C-bug)
    logictestbase.go:589: statement/query skipped with reason: unsupported configuration weak-iso-level-configs (no issue given)
    logictestbase.go:589: statement/query skipped with reason: unsupported configuration local-repeatable-read, statement/query only supports local-read-committed (no issue given)
    logictestbase.go:589: statement/query skipped with reason: unsupported configuration weak-iso-level-configs (no issue given)
    logictestbase.go:589: statement/query skipped with reason: unsupported configuration weak-iso-level-configs (no issue given)
    logic.go:3277: 
         
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share:234: BEGIN;
        SELECT * FROM t FOR UPDATE SKIP LOCKED
        expected:
            2
            
        but found (query options: "") :
            
[04:54:42] --- done: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share with config local-repeatable-read: 42 tests, 1 failures
    logic.go:4685: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share:240: error while processing
    logic.go:4685: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share:240: too many errors encountered, skipping the rest of the input
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestRepeatableReadLogic_select_for_share278845014
--- FAIL: TestRepeatableReadLogic_select_for_share (1.24s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=10</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRepeatableReadLogic_select_for_share.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62469

--- a/pkg/sql/logictest/testdata/logic_test/select_for_share
+++ b/pkg/sql/logictest/testdata/logic_test/select_for_share
@@ -214,6 +214,14 @@ CREATE TABLE t(a INT PRIMARY KEY);
 INSERT INTO t VALUES(1), (2);
 GRANT ALL ON t TO testuser;
 
+# Read both keys to ensure the INSERT's intents are resolved before any SKIP
+# LOCKED scan, which would skip unresolved intents (see #167582).
+query I rowsort
+SELECT * FROM t
+----
+1
+2
+
 user testuser
 
 statement ok

